src/includes/class-alma-wc-payment-gateway.php

```diff
@@ -285,17 +262,41 @@ protected function process_checkout_legal( $post_data ) {
 			return;
 		}
 
+		// By default, remove api consent
 		$this->settings['share_of_checkout_enabled_date'] = gmdate( 'Y-m-d' );
 		$value = 'no';
 
+		// Check if the live_api_key has changed. Remove the consent
 		if (
+			alma_wc_plugin()->settings->live_api_key !== $post_data['woocommerce_alma_live_api_key']
+		) {
+			$this->settings['share_of_checkout_enabled_date']             = '';
+			$this->settings['woocommerce_alma_share_of_checkout_enabled'] = '';
+		} elseif (
 			isset( $post_data['woocommerce_alma_share_of_checkout_enabled'] )
 			&& '1' == $post_data['woocommerce_alma_share_of_checkout_enabled']
 		) {
 			$value = 'yes';
 		}
 
 		$this->admin_helper_check_legal->send_consent( $value );
+
+		// Check if the mode has changed to live (by default in the module you are in test so the legal is not init).
+		if (
+			'live' === $post_data['woocommerce_alma_environment']
+			&& 'test' === alma_wc_plugin()->settings->get_environment()
```

Comment:
```suggestion
			$this->has_changed_to_live_env()
```
could be extracted in a helper if we don't want add a new method in this class